lib/apimatic-core/utilities/api_helper.rb

```diff
@@ -228,19 +226,62 @@ def self.clean_url(url)
       protocol + query + parameters
     end
 
+    # Deserialize the response based on the provided union_type.
+    # @param [UnionType] union_type The union type to validate and deserialize the response.
+    # @param [Object] response The response object to be deserialized.
+    # @param [Boolean] should_deserialize Flag indicating whether the response should be deserialized.
+    #                                    Default is true.
+    # @return [Object] The deserialized response based on the union_type.
+    def self.deserialize_union_type(union_type, response, should_symbolize = false, should_deserialize = false)
+      response = ApiHelper.json_deserialize(response, false, true) if should_deserialize
+
+      union_type_result = union_type.validate(response)
+
+      union_type_result.deserialize(response, should_symbolize: should_symbolize)
+    end
+
+    def self.apply_primitive_converter(value)
```

Comment:
Nits, Non-blocker: `apply_primitive_converter` method name should be renamed to `apply_primitive_type_parser`. Also add method documentation.

Reply:
done